app/javascript/components/providers/index/provider-index-component.vue

```diff
@@ -0,0 +1,141 @@
+
+    <div class="flex flex-col sm:flex-row flex-wrap justify-between w-full">
+      <div
+        class="flex items-start w-6/12"
+        v-for="element in this.providers"
+        :key="element.id"
+      >
+        <div class="flex flex-row items-start flex-none flex-grow-0 order-none w-full">
+          <provider-item
+            :provider="element"
+            @update="updateProvider"
+            @del="deleteProvider"
+          />
+        </div>
+      </div>
+    </div>
+
+    <base-modal
+      @ok="addProvider"
+      @cancel="toggleAddModal"
+      v-if="showingAdd"
+      :title="$t('msg.providers.add')"
+      :ok-button-label="$t('msg.add')"
+      :cancel-button-label="$t('msg.cancel')"
+    >
+      <providers-form
+        ref="addProviderInfo"
+      />
+    </base-modal>
+  </div>
+</template>
+
+<script>
+import camelcase from 'camelcase';
+import { getProviders, postProvider } from './../../../api/providers.js';
+import ProviderItem from './provider-item';
+import ProvidersForm from './providers-form';
+
+export default {
+  data() {
+    return {
+      showingDel: false,
+      showingAdd: false,
+      showingEdit: false,
+      providers: [],
+    };
+  },
+  components: {
+    ProviderItem,
+    ProvidersForm,
+  },
+
+  async created() {
+    try {
+      const response = await getProviders();
+      this.providers = response.data.data.map((element) => ({
+        id: element.id,
+        ...element.attributes,
+      }));
+      this.successResponse(response);
+    } catch (error) {
+      this.errorResponse(error);
+    }
+  },
+  methods: {
+    toggleAddModal() {
+      this.showingAdd = !this.showingAdd;
+    },
+
+    async addProvider() {
+      this.showingAdd = !this.showingAdd;
+      try {
+        const {
+          status,
+          data:
+            { data: { id, attributes },
+            },
+        } = await postProvider(this.$refs.addProviderInfo.form);
+        const providerToAdd = { id, ...attributes };
+        this.providers.push(providerToAdd);
+        this.successResponse(status);
+      } catch (error) {
+        this.errorResponse(error);
+      }
+    },
+    changeKeys(attributes) {
+      const newKeys = {};
+      const entries = Object.entries(attributes);
+      for (let i = 0; i < entries.length; i++) {
+        newKeys[camelcase(entries[i][0])] = entries[i][1];
+      }
+
+      return newKeys;
+    },
+
+    async updateProvider(res, providerId) {
+      this.showingEdit = !this.showingEdit;
+
+      const attributes = JSON.parse(res.config.data).provider;
+      const newAttributes = this.changeKeys(attributes);
+      const providerEdited = { id: providerId, ...newAttributes };
+      const objectIndex = this.providers.findIndex((obj => obj.id === providerId));
+      this.providers.splice(objectIndex, 1);
+      this.providers.splice(objectIndex, 0, providerEdited);
+    },
+
+    async deleteProvider(id, response) {
+      this.showingDel = !this.showingDel;
+      try {
```

Comment:
Lo mismo aquí, this.showingDel = !this.showingDel no está haciendo nada más que cambiar su valor, pero su valor nunca se usa

Reply:
Toda la razón!!